coordinator/src/dlc_protocol.rs

```diff
@@ -113,12 +116,57 @@ pub struct TradeParams {
     pub direction: Direction,
 }
 
+impl From<(ProtocolId, &commons::TradeParams)> for TradeParams {
+    fn from((protocol_id, trade_params): (ProtocolId, &commons::TradeParams)) -> Self {
+        Self {
+            protocol_id,
+            trader: trade_params.pubkey,
+            quantity: trade_params.quantity,
+            leverage: trade_params.leverage,
+            average_price: trade_params
+                .average_execution_price()
+                .to_f32()
+                .expect("to fit into f32"),
+            direction: trade_params.direction,
+        }
+    }
+}
+
 pub enum DlcProtocolState {
     Pending,
     Success,
     Failed,
 }
 
+#[derive(Clone, Debug)]
+pub enum DlcProtocolType {
+    Open { trade_params: TradeParams },
+    Renew { trade_params: TradeParams },
+    Settle { trade_params: TradeParams },
+    Close { trader: PublicKey },
+    ForceClose { trader: PublicKey },
+    Rollover { trader: PublicKey },
```

Comment:
I think it's okay.

I was worried because I thought we might run into data redundancy, but I think this enum is only used as an argument. And it is then mapped into a simpler enum to store the protocol name in the database.

Reply:
It's also used when reading from the database, to load additional data required if e.g. trade params should be attached to it.